packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -332,6 +328,13 @@ function NativeStackViewInner({ state, navigation, descriptors }: Props) {
           ? descriptors[previousKey]
           : undefined;
 
+        // workaround for rn-screens where gestureDirection has to be set on both
+        // current and previous screen - software-mansion/react-native-screens/pull/1509
+        const { gestureDirection } = descriptor.options;
+        if (gestureDirection && previousDescriptor) {
+          previousDescriptor.options.gestureDirection = gestureDirection;
```

Comment:
We can't mutate this since it's expected that a prop updating would trigger a re-render, however, a mutation wouldn't trigger a re-render.

Similar to `previousDescriptor`, you can check `nextDescriptor` and do:

```js
const nextGestureDirection = nextDescriptor?.options.gestureDirection;
const gestureDirection = nextGestureDirection != null ? nextGestureDirection : descriptor.options.gestureDirection;
```

Reply:
Yep, you're 100% right. Fixed in [07a1a54](https://github.com/react-navigation/react-navigation/pull/10674/commits/07a1a54625cfb8cf456f3da33c0af14ef7683956)